### PR TITLE
feat: migrate value parser to new style

### PR DESCRIPTION
**Pull request by powersj**
fixes: #11363

### CODE REVIEW

**Comment by powersj** [comment]
@srebhan I have two concerns about this one:

1) not sure about `MetricName` in this one? It is called in other tests, so I made it exported.
2) previous create parser, if `fieldName` was empty it was set to `value`. I have it set to that in the Parse function and init, but that doesn't seem right. Should it only be in the init and then ensure that init is called?

File: plugins/inputs/kafka_consumer/kafka_consumer_test.go [hunk 1]
```diff
@@ -271,8 +271,11 @@ func (c *FakeConsumerGroupClaim) Messages() <-chan *sarama.ConsumerMessage {
 
 func TestConsumerGroupHandler_Lifecycle(t *testing.T) {
 	acc := &testutil.Accumulator{}
-	parser := value.NewValueParser("cpu", "int", "", nil)
-	cg := NewConsumerGroupHandler(acc, 1, parser, testutil.Logger{})
+	parser := value.Parser{
+		MetricName: "cpu",
+		DataType:   "int",
+	}
+	cg := NewConsumerGroupHandler(acc, 1, &parser, testutil.Logger{})
```
Comment (srebhan) [inline plugins/inputs/kafka_consumer/kafka_consumer_test.go:278]: I think you should `Init` the parser here, same for the instances below.

File: plugins/parsers/value/parser.go [hunk 2]
```diff
@@ -9,16 +9,21 @@ import (
...
+	FieldName   string            `toml:"value_field_name"`
+	MetricName  string            `toml:"-"`
+	DefaultTags map[string]string `toml:"-"`
 }
 
-func (v *ValueParser) Parse(buf []byte) ([]telegraf.Metric, error) {
+func (v *Parser) Parse(buf []byte) ([]telegraf.Metric, error) {
+	if v.FieldName == "" {
+		v.FieldName = "value"
+	}
```
Comment (srebhan) [inline plugins/parsers/value/parser.go:25]: Please keep this in `Init`.

File: plugins/parsers/value/parser_test.go [hunk 3]
```diff
@@ -7,7 +7,10 @@ import (
 )
 
 func TestParseValidValues(t *testing.T) {
-	parser := NewValueParser("value_test", "integer", "", nil)
+	parser := Parser{
```
Comment (srebhan) [inline plugins/parsers/value/parser_test.go:10]: You miss the `Init`. Same for the instances below.

**Review by srebhan — COMMENTED** [review]
Thanks for working on this @powersj! I have some comments in the code, mostly about init, which brings me to your questions:

> not sure about MetricName in this one? It is called in other tests, so I made it exported.

That's fine.

> previous create parser, if fieldName was empty it was set to value. I have it set to that in the Parse function and init, but that doesn't seem right. Should it only be in the init and then ensure that init is called?

Exactly. [Telegraf will init all "RunningParsers" during startup](https://github.com/powersj/telegraf/blob/412c82a8ffd59f218d5f8a644af67e3b36b649b7/agent/agent.go#L200). If you get an error with `config_test` you probably need [this patch](https://github.com/influxdata/telegraf/pull/11405/commits/4cf589da603c37906f85445aefec4212342cd008)...

**Comment by powersj** [comment]
Thanks @srebhan for the review and tips. I think this one is ready to go

**Review by srebhan — APPROVED** [review]
Looks good to me. Thanks @powersj!

**Pull request merged**